eng/testing/RunnerTemplate.sh

```diff
@@ -220,51 +190,69 @@ if [[ $test_exitcode -ne 0 ]]; then
   echo ulimit -c value: $(ulimit -c)
 fi
 
-if [[ "$(uname -s)" == "Linux" && $test_exitcode -ne 0 ]]; then
-  if [ -n "$HELIX_WORKITEM_PAYLOAD" ]; then
-
-    # For abrupt failures, in Helix, dump some of the kernel log, in case there is a hint
-    if [[ $test_exitcode -ne 1 ]]; then
-      dmesg | tail -50
-    fi
-
-    have_sleep=$(which sleep)
-    if [ -x "$have_sleep" ]; then
-      echo Waiting a few seconds for any dump to be written..
-      sleep 10s
-    fi
-  fi
-
+if [[ $system_name == "Linux" && $test_exitcode -ne 0 ]]; then
   echo cat /proc/sys/kernel/core_pattern: $(cat /proc/sys/kernel/core_pattern)
   echo cat /proc/sys/kernel/core_uses_pid: $(cat /proc/sys/kernel/core_uses_pid)
   echo cat /proc/sys/kernel/coredump_filter: $(cat /proc/sys/kernel/coredump_filter)
 
-  echo Looking around for any Linux dump..
-
   # Depending on distro/configuration, the core files may either be named "core"
   # or "core.<PID>" by default. We read /proc/sys/kernel/core_uses_pid to
   # determine which it is.
   core_name_uses_pid=0
   if [[ -e /proc/sys/kernel/core_uses_pid && "1" == $(cat /proc/sys/kernel/core_uses_pid) ]]; then
     core_name_uses_pid=1
   fi
+  
+  # The osx dumps are too large to egress the machine
+  echo Looking around for any Linux dumps...
 
   if [[ "$core_name_uses_pid" == "1" ]]; then
     # We don't know what the PID of the process was, so let's look at all core
     # files whose name matches core.NUMBER
-    echo Looking for files matching core.* ...
-    for f in core.*; do
-      [[ $f =~ core.[0-9]+ ]] && print_info_from_core_file "$f" "dotnet" && copy_core_file_to_temp_location "$f" && rm "$f"
+    echo "Looking for files matching core.* ..."
+    for f in $(find . -name "core.*"); do
+      [[ $f =~ core.[0-9]+ ]] && move_core_file_to_temp_location "$f"
     done
-  elif [ -f core ]; then
-    echo found a dump named core in $EXECUTION_DIR !
-    print_info_from_core_file "core" "dotnet"
-    copy_core_file_to_temp_location "core"
-    rm "core"
-  else
-    echo ... found no dump in $PWD
+  fi
+
+  if [ -f core ]; then
+    move_core_file_to_temp_location "core"
   fi
 fi
+
+if [ -n "$HELIX_WORKITEM_PAYLOAD" ]; then
+  # For abrupt failures, in Helix, dump some of the kernel log, in case there is a hint
+  if [[ $test_exitcode -ne 1 ]]; then
+    dmesg | tail -50
+  fi
+
+  have_sleep=$(which sleep)
+  if [ -x "$have_sleep" ]; then
+    echo Waiting a few seconds for any dump to be written...
+    sleep 10s
+  fi
+fi
+
+if [[ -z "$__IsXUnitLogCheckerSupported" ]]; then
+  echo "The '__IsXUnitLogCheckerSupported' env var is not set."
+elif [[ "$__IsXUnitLogCheckerSupported" != "1" ]]; then
+  echo "XUnitLogChecker not supported for this test case. Skipping."
+else
+  echo ----- start ===============  XUnitLogChecker Output =====================================================
+  
+  invoke_xunitlogchecker "$HELIX_DUMP_FOLDER"
+
+  # osx stores giant dumps in a separate folder
```

Comment:
This is only the system cores - which I thought we disabled?

Reply:
It seems that $HELIX_DUMP_FOLDER equals `/cores` anyway in OSX, which means this code is unnecessary, yes, because `invoke_xunitlogchecker` is called twice on the same folder. So yes, I can remove it.

Reply:
I opened this issue to track why the /cores dumps are not being read by XUnitLogChecker: https://github.com/dotnet/runtime/issues/94720